test/relayHub/RelayHubContractScenarios.test.ts

```diff
@@ -0,0 +1,737 @@
+import {ERR_NOT_OWNER, ERR_UNSTAKED, oneRBTC} from "../utils";
+import chai, {assert, expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import {mine} from "@nomicfoundation/hardhat-network-helpers";
+import {ethers} from "hardhat";
+import {FakeContract, MockContract, smock} from "@defi-wonderland/smock";
+import {Penalizer, RelayHub, SmartWallet, SmartWalletFactory} from "../../typechain-types";
+import {createContractDeployer} from "./utils";
+import {SignerWithAddress} from "@nomiclabs/hardhat-ethers/signers";
+import {
+  getLocalEip712Signature,
+  TypedRequestData,
+  RelayRequest,
+  ForwardRequest,
+  RelayData, TypedDeployRequestData, getLocalEip712DeploySignature
+} from '../utils/EIP712Utils';
+import {Wallet} from "ethers";
+import {EnvelopingTypes, IForwarder} from 'typechain-types/contracts/RelayHub';
+
+type DeployRequest = EnvelopingTypes.DeployRequestStruct;
+
+chai.use(smock.matchers);
+chai.use(chaiAsPromised);
+
+const ZERO_ADDRESS = ethers.constants.AddressZero;
+
+describe('RelayHub contract - Manager related scenarios', function () {
+  let deployRelayHubContract: ReturnType<typeof createContractDeployer>;
+  let relayManager: SignerWithAddress, relayManagerAddr: string;
+  let relayWorker: SignerWithAddress, relayWorkerAddr: string;
+  let relayOwner: SignerWithAddress, relayOwnerAddr: string;
+  let otherUsers: SignerWithAddress[];
+
+  let fakePenalizer: FakeContract<Penalizer>;
+  let mockRelayHub: MockContract<RelayHub>;
+
+
+  beforeEach(async function () {
+    [relayOwner, relayManager, relayWorker, ...otherUsers] = await ethers.getSigners();
+    relayManagerAddr = relayManager.address;
+    relayWorkerAddr = relayWorker.address;
+    relayOwnerAddr = relayOwner.address;
+
+    fakePenalizer = await smock.fake('Penalizer');
+  });
+
+  afterEach(function () {
+    mockRelayHub = undefined as unknown as MockContract<RelayHub>;
+  });
+
+  describe('Manager tests - Stake related scenarios', function() {
```

Comment:
This whole describe is duplicated in the RelayHub.test.ts file.

Reply:
Removed duplicated